test/example/SnapshotGateTest.js

```diff
@@ -0,0 +1,717 @@
+  mockVoucherInitValues,
+  mockSeller,
+  mockCondition,
+  accountId,
+} = require("../util/mock");
+const { oneWeek, oneMonth, maxPriorityFeePerGas } = require("../util/constants");
+const { deploySnapshotGateExample } = require("../../scripts/example/SnapshotGate/deploy-snapshot-gate");
+const { deployMockTokens } = require("../../scripts/util/deploy-mock-tokens");
+const { getEvent } = require("../util/utils");
+
+/**
+ *  Test the SnapshotGate example contract
+ */
+describe("SnapshotGate", function () {
+  // Common vars
+  let deployer,
+    pauser,
+    operator,
+    operator2,
+    admin,
+    clerk,
+    treasury,
+    rando,
+    operatorDR,
+    adminDR,
+    clerkDR,
+    treasuryDR,
+    protocolTreasury,
+    bosonToken,
+    holder1,
+    holder2,
+    holder3,
+    holder4,
+    holder5;
+  let protocolDiamond, accessController, accountHandler, offerHandler, groupHandler;
+  let offerId, seller, seller2, disputeResolverId;
+  let price, foreign20;
+  let protocolFeePercentage, protocolFeeFlatBoson, buyerEscalationDepositPercentage;
+  let disputeResolver, disputeResolverFees;
+  let snapshotGate;
+  let groupId, offerIds, condition, group, groups;
+  let voucherInitValues;
+  let emptyAuthToken;
+  let sellerId, agentId;
+  let offer, offers, otherSellerOfferId;
+  let offerDates, offerDurations;
+  let snapshot, snapshotTokenSupplies, snapshotTokenCount, holders, holderByAddress;
+
+  beforeEach(async function () {
+    // Make accounts available
+    [
+      deployer,
+      pauser,
+      admin,
+      treasury,
+      rando,
+      adminDR,
+      treasuryDR,
+      protocolTreasury,
+      operator2,
+      bosonToken,
+      holder1,
+      holder2,
+      holder3,
+      holder4,
+      holder5,
+    ] = await ethers.getSigners();
+
+    // make all account the same
+    operator = clerk = admin;
+    operatorDR = clerkDR = adminDR;
+
+    // Deploy the Protocol Diamond
+    [protocolDiamond, , , , accessController] = await deployProtocolDiamond(maxPriorityFeePerGas);
+
+    // Temporarily grant UPGRADER role to deployer account
+    await accessController.grantRole(Role.UPGRADER, deployer.address);
+
+    // Grant PROTOCOL role to ProtocolDiamond address and renounces admin
+    await accessController.grantRole(Role.PROTOCOL, protocolDiamond.address);
+
+    // Temporarily grant PAUSER role to pauser account
+    await accessController.grantRole(Role.PAUSER, pauser.address);
+
+    // Cut the protocol handler facets into the Diamond
+    await deployProtocolHandlerFacets(
+      protocolDiamond,
+      [
+        "AccountHandlerFacet",
+        "AgentHandlerFacet",
+        "SellerHandlerFacet",
+        "BuyerHandlerFacet",
+        "DisputeResolverHandlerFacet",
+        "ExchangeHandlerFacet",
+        "OfferHandlerFacet",
+        "FundsHandlerFacet",
+        "DisputeHandlerFacet",
+        "TwinHandlerFacet",
+        "BundleHandlerFacet",
+        "GroupHandlerFacet",
+        "PauseHandlerFacet",
```

Comment:
Do we need all of these?  Since this is an example, we should be specific to show how to test such a gate properly.

Reply:
fixed